sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/codegen/GenerateMutableProjection.scala

```diff
@@ -45,7 +45,32 @@ object GenerateMutableProjection extends CodeGenerator[Seq[Expression], () => Mu
           else
             mutableRow.${ctx.setColumn(e.dataType, i, evaluationCode.primitive)};
         """
-    }.mkString("\n")
+    }
+    val partitionedProjectionCode = projectionCode.foldLeft(List.empty[String]) {
+      (acc, code) =>
+        acc match {
+          case Nil => List(code)
+          case head::tail =>
+            // code size limit is 64kb and each char takes less or equal to 2 bytes
+            if (head.length < 32 * 1000) {
+              s"$head\n$code"::tail
+            } else {
+              code::acc
+            }
+        }
+    }
+      .zipWithIndex
+      .map {
+        case (body, i) =>
+          s"""
+             private void apply$i(InternalRow i) {
+               $body
+             }
+           """
+      }
+    val projectionCalls = ((partitionedProjectionCode.length - 1) to 0 by -1)
+      .map(i => s"apply$i(i);")
```

Comment:
As we have some discussion on #7418 probably it's too heavy to call function for getting the one field value. Particularly for the case:
`select a+b from src`